src/shogun/distributions/classical/GaussianDistributionNew.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2013 Heiko Strathmann
+ */
+
+#include <shogun/distributions/classical/GaussianDistributionNew.h>
+
+#ifdef HAVE_EIGEN3
+
+#include <shogun/base/Parameter.h>
+#include <shogun/mathematics/eigen3.h>
+
+using namespace shogun;
+using namespace Eigen;
+
+CGaussianDistributionNew::CGaussianDistributionNew() : CProbabilityDistribution()
+{
+	init();
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		float64_t cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	init();
+
+	m_mean=mean;
+
+	m_cov=SGMatrix<float64_t>(1,1);
+	m_cov(1,1)=cov;
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		SGVector<float64_t> cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	REQUIRE(mean.vlen==cov.vlen, "Mean must have same dimension as "
+			"covariance, which is %dx%d, but is %d\n",
+			cov.vlen, cov.vlen, mean.vlen);
+
+	init();
+
+	m_mean=mean;
+
+	m_cov=cov.convert_to_matrix(cov,1,cov.vlen,false);
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		SGMatrix<float64_t> cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	REQUIRE(cov.num_rows==cov.num_cols, "Covariance must be square but is "
+			"%dx%d\n", cov.num_rows, cov.num_cols);
+	REQUIRE(mean.vlen==cov.num_cols, "Mean must have same dimension as "
+			"covariance, which is %dx%d, but is %d\n",
+			cov.num_rows, cov.num_cols, mean.vlen);
+
+	init();
+
+	m_mean=mean;
+
+	if (!cov_is_cholesky)
+	{
+		Map<MatrixXd> eigen_cov(cov.matrix, cov.num_rows, cov.num_cols);
+		m_cov=SGMatrix<float64_t>(cov.num_rows, cov.num_cols);
+		Map<MatrixXd> eigen_L(m_cov.matrix, m_cov.num_rows, m_cov.num_cols);
+
+		/* compute cholesky */
+		LLT<MatrixXd> llt(eigen_cov);
+		if (llt.info()==NumericalIssue)
+		{
+			/* try to compute smalles eigenvalue for information */
+			SelfAdjointEigenSolver<MatrixXd> solver(eigen_cov);
+			if (solver.info() == Success)
+			{
+				VectorXd ev=solver.eigenvalues();
+				SG_ERROR("Error computing Cholesky of Gaussian's covariance. "
```

Comment:
The error is only printed if Eigenvalue computation was a success. 
I suggest to print the smallest eigenvalue in a warning, and then throw the error afterwards, independent of the eigenvalue solver